app/assets/javascripts/repository_columns/index.js

```diff
@@ -0,0 +1,224 @@
+        modalHtml.modal('show');
+        initDeleteSubmitAction(modalHtml, $(modalHtml.find('form')));
+      }).fail(function() {
+        HelperModule.flashAlertMsg(
+          I18n.t('libraries.repository_columns.no_permissions'), 'danger'
+        );
+      });
+    });
+  }
+
+  function insertNewListItem(column) {
+    var html = `<li class="list-group-item row" data-id="${column.id}"
+                    data-destroy-url="${column.destroy_html_url}"
+                    data-edit-url="${column.edit_url}">
+                  <div class="col-xs-8">
+                    <span class="pull-left column-name">${column.name}</span>
+                  </div>
+                  <div class="col-xs-4">
+                    <span class="controlls pull-right">
+                      <button class="btn btn-default edit-repo-column" data-action="edit">
+                        <span class="fas fa-pencil-alt"></span>&nbsp;
+                        <%= I18n.t "libraries.repository_columns.index.edit_column" %>
+                      </button>&nbsp;
+                      <button class="btn btn-default delete-repo-column" data-action="destroy">
+                        <span class="fas fa-trash-alt"></span>&nbsp;
+                        <%= I18n.t "libraries.repository_columns.index.delete_column" %>
+                      </button>
+                    </span>
+                  </div>
+                </li>`;
+
+    // remove element if already persent
+    $('[data-id="' + column.id + '"]').remove();
+    $(html).insertBefore('.repository-columns-body ul li:first');
+    // remove 'no column' list item
+    $('[data-attr="no-columns"]').remove();
+  }
+
+  function initNewColumnModal() {
+    var modalID = '#manageRepositoryColumn';
+    $('.repository-columns-header').off('click', '#new-repo-column-modal').on('click', '#new-repo-column-modal', function() {
+      var modalUrl = $(this).attr('data-modal-url');
+      $.get(modalUrl, function(data) {
+        $(modalID).find('.modal-content').html(data.html);
+        $(modalID).modal('show');
+
+        initColumnTypeSelector();
+        $('[data-column-type="RepositoryTextValue"]').show();
+        setTimeout(function() {
+          $('#repository_column_name').focus();
+        }, 500);
+
+        $('#new-repo-column-submit').on('click', function() {
+          var url = $('#repository_column_data_type').find(':selected').data('create-url');
+          $.ajax({
+            url: url,
+            data: { repository_column: { name: $('#repository_column_name').val() } },
+            type: 'POST',
+            success: function(data2) {
+              insertNewListItem(data2);
+              HelperModule.flashAlertMsg(data2.message, 'success');
+              if (modalID) {
+                $(modalID).modal('hide');
+              }
+            },
+            error: function() {
+            }
+          });
+        });
+      });
+    });
+  }
```

Comment:
Can you simplify it split for 2 functions. 
1) initNewColumnModal
2) initNewSubmitAction
Like we have for delete